lib/utils/checkAgainstRule.js

```diff
@@ -3,39 +3,45 @@
 const normalizeRuleSettings = require('../normalizeRuleSettings');
 const Result = require('postcss/lib/result');
 const rules = require('../rules');
+const { isPlainObject } = require('./validateTypes');
 
 /**
  * Useful for third-party code (e.g. plugins) to run a PostCSS Root
  * against a specific rule and do something with the warnings
  * @template T
  * @template {Object} O
  * @param {{
-		ruleName: string,
-		ruleSettings: import('stylelint').ConfigRuleSettings<T, O>,
-		root: import('postcss').Root,
-	}} options
+ *   ruleName: string,
+ *   ruleSettings: import('stylelint').ConfigRuleSettings<T, O>,
+ *   root: import('postcss').Root,
+ *   result?: import('stylelint').PostcssResult,
+ * }} options
  * @param {(warning: import('postcss').Warning) => void} callback
  * @returns {void}
  */
 function checkAgainstRule(options, callback) {
-	if (!options)
-		throw new Error(
-			"checkAgainstRule requires an options object with 'ruleName', 'ruleSettings', and 'root' properties",
-		);
+	if (!isPlainObject(options)) throw new Error('checkAgainstRule requires an options object');
```

Comment:
[suggest] Thanks for improving the error messages. In addition, could you refactor them as follows?

- Remove the `checkAgainstRule` word from the messages because an error raised in `checkAgainstRule` is explicit from a stack trace
- Use `"` (double) instead of `'` (single) in the messages for consistency because we often have used `"` in this project

E.g.

```diff
-throw new Error("checkAgainstRule requires a 'ruleName' option");
+throw new Error('require a "ruleName" option');
```

Reply:
Agreed, there's a clear opportunity to remove the redundancy 👍 Expanding on the suggestion, what do you think of the following messages:
```js
throw new Error('An options object is required')
throw new Error('A callback function is required')
throw new Error('The "ruleName" option is required')
throw new Error('The "ruleSettings" option is required')
throw new Error('The "root" option is required')
```
Or following the rule/report conventions:
```js
throw new Error('Expected an options object')
throw new Error('Expected a callback function')
throw new Error('Expected a "ruleName" option')
throw new Error('Expected a "ruleSettings" option')
throw new Error('Expected a "root" option')
```

Reply:
Nice suggestion. I prefer the latter, but both are acceptable. I leave it to you to decide.

Reply:
The latter it is! https://github.com/stylelint/stylelint/pull/6460/commits/b5636bd966adeda34c6a9d52b7a2dbc137e0b5b9